tests/foreman/cli/test_remoteexecution.py

```diff
@@ -254,39 +246,35 @@ class RemoteExecutionTestCase(CLITestCase):
     """Implements job execution tests in CLI."""
 
     @classmethod
-    @skip_if_not_set('clients', 'fake_manifest')
+    @skip_if_not_set('clients', 'fake_manifest', 'vlan_networking')
     def setUpClass(cls):
         """Create Org, Lifecycle Environment, Content View, Activation key
         """
         super(RemoteExecutionTestCase, cls).setUpClass()
-        cls.org = make_org()
+        cls.org = entities.Organization().create()
         ssh.command(
             '''echo 'getenforce' > {0}'''.format(TEMPLATE_FILE)
         )
-        cls.env = make_lifecycle_environment({
-            u'organization-id': cls.org['id'],
-        })
-        cls.content_view = make_content_view({
-            u'organization-id': cls.org['id'],
-        })
-        cls.activation_key = make_activation_key({
-            u'lifecycle-environment-id': cls.env['id'],
-            u'organization-id': cls.org['id'],
-        })
-        # Add subscription to Satellite Tools repo to activation key
-        setup_org_for_a_rh_repo({
-            u'product': PRDS['rhel'],
-            u'repository-set': REPOSET['rhst7'],
-            u'repository': REPOS['rhst7']['name'],
-            u'organization-id': cls.org['id'],
-            u'content-view-id': cls.content_view['id'],
-            u'lifecycle-environment-id': cls.env['id'],
-            u'activationkey-id': cls.activation_key['id'],
-        })
+        # create subnet for current org, default loc and domain
+        cls.sn = entities.Subnet(
+            domain=[1],
+            gateway=settings.vlan_networking.gateway,
+            ipam='DHCP',
+            location=[DEFAULT_LOC_ID],
+            mask=settings.vlan_networking.netmask,
+            network=settings.vlan_networking.subnet,
+            organization=[cls.org.id]
+        ).create()
```

Comment:
yeah, this is supposed to test package installation using remote execution feature. The product/repo part is completely unrelated and is there just as a setup (to provide something that can be installed). I could actually do the setup completely outside of the satellite by simply create a custom repo file directly on the host.

All that being said, there is no reason of using the same endpoint. API is the fastest and cleanest solution here. So i'm simply aiming to get to the actual testing by the fastest and simplest path possible (performance increase - check).

> if tomorrow entities.Subnet.create() starts failing why entire CLI module should fail too?

can't spot any argument in this, as i can claim the same about `cli.Subnet.create()` - plus, the `cli.Subnet` actually provide more possible points of failure than the plain api (cli command boils down to the actual api call, so they either fail both, or there's a bug in nailgun which is ..well, maintained by us)

Reply:
API is used to speedup things like we do for UI, setup done by nailgun and own testing continues in UI. 

Lets say if we have in setup a  product creation then it makes no sense to create it by CLI/UI since we have CLI/UI already covered by other tests - CLI/UI Product creation test cases. 

Therefore  the argument saying that by using API in setups we skip something is not valid, IMO.

Reply:
Personally I see no evil in API setup as these steps already covered in corresponding CLI tests. Though I must admit that it is a little bit inconsistent with other CLI tests :)

Reply:
It is obvious that API is a better solution for setup entries for UI as it slowest interface possible, but here I don't see any reason to use it. You introduce redundant dependencies and ruin few key isolation principles just for a little bit cleaner code and none performance increase (I don't count some seconds as serious increase)
At that moment we have huge code base and it will be harder and harder to manage it. And as more approaches we have the more impossible to find the root cause for the problems.
So if we like to start using any new approach, let's create wiki with details, discuss and start using it incrementally through the code.
But at that moment I am strongly NACK any chaotic code changes in that direction. Even now we have too many of them, so I vote for slower, worse approach, but consistent one

Reply:
>cli command boils down to the actual api call, so they either fail both, or there's a bug in nailgun

Do you want me to find you a dozen examples when this statement is wrong? 🙂 i'm not sure of exact root causes - sometimes api stops returning required fields for us, sometimes url for api call changes, doesn't matter, the result is our cli commands continue working while api stopped.
And i don't think small performance increase is a good enough reason for decreasing isolation and basically messing different parts of our code together, as we aren't talking about ui tests, where api calls can be 10 times faster.
But that's my vision, let's discuss. This may be a good topic for our friday's auto meeting.

Reply:
uhm..i just actually stumbled upon a problem - there seems not to be a way of assigning a rex-capsule to a subnet using hammer :D I honestly wanted to make you happy.

Could I use this as an excuse for using api here and as a bonus i'd attach a pic of a cute panda?

![img](https://cdn.vectorstock.com/i/thumb-large/14/89/little-cute-panda-isolated-on-white-hand-vector-7931489.jpg)

Reply:
As i stated above,
>The only excuses are [...] the cases when interface doesn't support some operation and there's no other way than use different interface.

So it's acceptable for that specific subnet creation if there's really no other way. Btw, do we have a BZ for that?

<details>
  <summary>P.S.</summary>
that's a cute panda 🙂 
</details>

Reply:
@abalakh yeah, here's one: https://bugzilla.redhat.com/show_bug.cgi?id=1370460